OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseUnitWebViewFragment.java

```diff
@@ -331,15 +343,9 @@ private void showShiftDateSnackBar(boolean isSuccess) {
     private void updateCalendarEvents() {
         trackCalendarEvent(Analytics.Events.CALENDAR_SYNC_UPDATE, Analytics.Values.CALENDAR_SYNC_UPDATE);
         long newCalId = CalendarUtils.createOrUpdateCalendar(getContextOrThrow(), accountName, CalendarContract.ACCOUNT_TYPE_LOCAL, calendarTitle);
-        if (courseDateViewModel.getCourseDates().getValue() != null) {
-            for (CourseDateBlock courseDateBlock : courseDateViewModel.getCourseDates().getValue().getCourseDateBlocks()) {
-                ConfigUtil.Companion.checkCalendarSyncEnabled(environment.getConfig(), response ->
-                        CalendarUtils.addEventsIntoCalendar(getContextOrThrow(), newCalId, unit.getCourseId(),
-                                courseName, courseDateBlock, response.isDeepLinkEnabled()));
-            }
-            showCalendarUpdatedSnackbar();
-            trackCalendarEvent(Analytics.Events.CALENDAR_UPDATE_SUCCESS, Analytics.Values.CALENDAR_UPDATE_SUCCESS);
-        }
+        ConfigUtil.Companion.checkCalendarSyncEnabled(environment.getConfig(), response ->
```

Comment:
There is one change in implementation here. previously if course dates values were null we weren't tracking analytics but in the new implementation (viewModel) even if the values are null we are still sending analytics. 
Is this change intended?